src/Symfony/Component/Translation/Translator.php

```diff
@@ -193,12 +193,20 @@ public function getFallbackLocales(): array
     /**
      * {@inheritdoc}
      */
-    public function trans(?string $id, array $parameters = [], string $domain = null, string $locale = null)
+    public function trans($id, array $parameters = [], string $domain = null, string $locale = null)
     {
+        if ($id instanceof Translatable) {
+            $parameters += $id->getParameters();
+            $domain = $id->getDomain();
+            $id = $id->getMessageId();
+        }
+
         if (null === $id || '' === $id) {
             return '';
         }
 
+        $id = (string) $id;
+
         if (null === $domain) {
             $domain = 'messages';
```

Comment:
this should also be kept in the else case,  if $id is not a translatable. Otherwise we implicitly override it...

i guess we need a bunch of unit tests ;)

Reply:
No longer needed (except tests, those will come once we get functionality nailed down).